### PR TITLE
GitHub workflows

### LINKED ISSUE #102: Test failures on python3.8/python3.9
FYI: I see one test failure with the latest release on python3.8 and python3.9.

```
============================= test session starts ==============================
platform linux -- Python 3.9.1, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /build/source
plugins: cov-2.10.1
collected 1117 items                                                           

tests/test_base.py ....                                                  [  0%]
tests/test_composites.py ....                                            [  0%]
tests/test_definitions.py ...                                            [  0%]
tests/test_fields.py ..................                                  [  2%]
tests/test_json_schema.py .............................................. [  6%]
........................................................................ [ 13%]
........................................................................ [ 19%]
........................................................................ [ 26%]
........................................................................ [ 32%]
........................................................................ [ 38%]
........................................................................ [ 45%]
........................................................................ [ 51%]
........................................................................ [ 58%]
........................................................................ [ 64%]
........................................................................ [ 71%]
........................................................................ [ 77%]
........................................................................ [ 84%]
........................................................................ [ 90%]
......................................................................F  [ 96%]
tests/test_schemas.py ........................                           [ 99%]
tests/tokenize/test_tokenize_json.py .....                               [ 99%]
tests/tokenize/test_tokenize_yaml.py ....                                [ 99%]
tests/tokenize/test_validate_json.py .                                   [ 99%]
tests/tokenize/test_validate_yaml.py .                                   [100%]

=================================== FAILURES ===================================
________________ test_to_json_schema_complex_regular_expression ________________

    def test_to_json_schema_complex_regular_expression():
        field = typesystem.String(pattern=re.compile("foo", re.IGNORECASE | re.VERBOSE))
        with pytest.raises(ValueError) as exc_info:
            to_json_schema(field)

        expected = (
            "Cannot convert regular expression with non-standard flags "
            "to JSON schema: RegexFlag."
        )
>       assert str(exc_info.value).startswith(expected)
E       AssertionError: assert False
E        +  where False = <built-in method startswith of str object at 0x7ffff5e2a850>('Cannot convert regular expression with non-standard flags to JSON schema: RegexFlag.')
E        +    where <built-in method startswith of str object at 0x7ffff5e2a850> = 'Cannot convert regular expression with non-standard flags to JSON schema: re.IGNORECASE|re.UNICODE|re.VERBOSE'.startswith
E        +      where 'Cannot convert regular expression with non-standard flags to JSON schema: re.IGNORECASE|re.UNICODE|re.VERBOSE' = str(ValueError('Cannot convert regular expression with non-standard flags to JSON schema: re.IGNORECASE|re.UNICODE|re.VERBOSE'))
E        +        where ValueError('Cannot convert regular expression with non-standard flags to JSON schema: re.IGNORECASE|re.UNICODE|re.VERBOSE') = <ExceptionInfo ValueError('Cannot convert regular expression with non-standard flags to JSON schema: re.IGNORECASE|re.UNICODE|re.VERBOSE') tblen=2>.value

tests/test_json_schema.py:172: AssertionError
```

### PR DESCRIPTION
**Pull request by aminalaee**
Not changing any logics, only adding github workflows like other repos.

#Fixes #102

### CODE REVIEW

File: scripts/check [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export SOURCE_FILES="typesystem tests"
+
+set -x
+
+export SOURCE_FILES="typesystem tests"
+
+${PREFIX}isort --check --diff --project=typesystem $SOURCE_FILES
+${PREFIX}black --check --diff $SOURCE_FILES
+${PREFIX}flake8 $SOURCE_FILES
+# ${PREFIX}mypy $SOURCE_FILES
```
Comment (Kludex) [inline scripts/check:16]: You can _uncomment_ and add the same `files` as `uvicorn`: https://github.com/encode/uvicorn/blob/ad761362b49278606520a1ae4712f68c8016cfb5/setup.cfg#L5-L44

If you want.

Comment (aminalaee) [reply scripts/check]: @Kludex thanks for that. Is there anything else you think should be changed?

**Review by florimondmanca — APPROVED** [review]

**Comment by florimondmanca** [comment]
Only reviewing as a bystander, overall this seems in line with other projects indeed. I'll let others give a review as well.

File: .github/workflows/test-suite.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    branches: ["master"]
+
+jobs:
+  tests:
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: "ubuntu-latest"
+
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10.0-beta.3"]
```
Comment (Kludex) [inline .github/workflows/test-suite.yml:17]:
```suggestion
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10.0-rc.1"]
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**